packages/expo-audio/src/index.ts

```diff
@@ -40,10 +40,7 @@ export function useAudioSampleListener(
   player.setAudioSamplingEnabled(true);
   useEffect(() => {
     const subscription = player.addListener('onAudioSampleUpdate', listener);
-    return () => {
-      player.setAudioSamplingEnabled(false);
-      subscription.remove();
-    };
```

Comment:
This is fine on android but causes a crash on iOS because the SharedObject has already been destroyed